xlators/cluster/dht/src/dht-rebalance.c

```diff
@@ -3612,13 +3605,20 @@ gf_defrag_fix_layout(xlator_t *this, gf_defrag_info_t *defrag, loc_t *loc,
                    "lookup failed for:%s", loc->path);
 
             defrag->total_failures++;
-            goto out;
+
+            if (conf->decommission_in_progress) {
+                defrag->defrag_status = GF_DEFRAG_STATUS_FAILED;
+                ret = -1;
+                goto out;
```

Comment:
mv `goto out;` outside this block for getting out in any case

Reply:
Done